src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -68,9 +72,9 @@ internal PdbLogger(bool logging)
             }
         }
 
-        private void MaybeFlush()
+        private void EnsureSpace(int space)
         {
-            if (_logData.Count >= bufferFlushLimit)
+            if (_logData.Count + space >= bufferFlushLimit)
```

Comment:
It should not? What if there is a value (argument to the PDB writer) that is longer than bufferFlushLimit (that is space > bufferFlushLimit)? I don't see code that splits it into multiple pieces. I think that's fine this is already handled by the builder.